nvflare/private/fed/client/client_executor.py

```diff
@@ -317,24 +317,36 @@ def send_aux_command(self, shareable: Shareable, run_number):
 
     def abort_train(self, client, run_number):
         with self.lock:
-            process_status = self.run_processes.get(run_number, {}).get(RunProcessKey.STATUS, ClientStatus.NOT_STARTED)
-            if process_status == ClientStatus.STARTED:
-                try:
-                    child_process = self.run_processes[run_number][RunProcessKey.CHILD_PROCESS]
-                    conn_client = self.get_conn_client(run_number)
-                    if conn_client:
-                        data = {"command": AdminCommandNames.ABORT, "data": {}}
-                        conn_client.send(data)
-                        self.logger.debug("abort sent")
-
-                    threading.Thread(target=self._terminate_process, args=[child_process, run_number]).start()
-                except Exception as e:
-                    self.logger.error(f"abort_train execution exception: {e}.", exc_info=True)
-                finally:
-                    if conn_client:
-                        conn_client.close()
-                    self.run_processes.pop(run_number)
-                    self.cleanup()
+            retry = 1
+            while retry >= 0:
+                process_status = self.run_processes.get(run_number, {}).get(
+                    RunProcessKey.STATUS, ClientStatus.NOT_STARTED
+                )
+                if process_status == ClientStatus.STARTED:
+                    try:
+                        child_process = self.run_processes[run_number][RunProcessKey.CHILD_PROCESS]
+                        conn_client = self.get_conn_client(run_number)
+                        if conn_client:
+                            data = {"command": AdminCommandNames.ABORT, "data": {}}
+                            conn_client.send(data)
+                            self.logger.debug("abort sent")
+
+                        threading.Thread(target=self._terminate_process, args=[child_process, run_number]).start()
+                        break
+                    except Exception as e:
+                        if retry == 0:
+                            self.logger.error(
+                                f"abort_train execution exception: {e} for run: {run_number}.", exc_info=True
+                            )
+                        retry -= 1
+                        time.sleep(5.0)
```

Comment:
retry probably exponential delay 
i.e. each sleep is longer than previous one 
for example
    sleep time = 5*2^k-1
https://en.wikipedia.org/wiki/Exponential_backoff

Reply:
done